src/public/js/rs.js

```diff
@@ -2,23 +2,59 @@
 var RS = (function() {
   // (actually ROT-13 right now, as a dummy example. We can replace w/ RS)
   var ROT = 13;
+  var GEN = 3;
+  var GF = 256;
 
   function encode(msg) {
     // dummy cipher for now -- advances each letter in message by ROT
-    return shift(msg, ROT);
+    return UTIL.shift(msg, ROT);
   }
 
   function decode(msg) {
     // dummy cipher for now -- advances each letter in message by -ROT
-    return shift(msg, -ROT);
+    return UTIL.shift(msg, -ROT);
   }
 
-  function shift(msg, rot) {
-    return msg.split('').reduce(function(accumulator, letter, index) {
-      var shiftedCharCode = letter.charCodeAt() + rot;
-      var shiftedLetter = String.fromCharCode(shiftedCharCode);
-      return accumulator + shiftedLetter;
-    }, '');
+  // Encodes using systematic BCH Reed-Solomon on Galois Field 256
+  // Message should be passed in as length k block
+  // k specifies message length
+  // n specifies block length (number of symbols transmitted)
+  // n - k is the number of code symbols added.
+  // Encoding procedure has the largest order coefficient on the left
+  // and the smallest on the right. IE: [1,2,3] -> x^2 + 2x + 3
+  // The largest order coeeficient is the [0] element of msg so as
+  // to maintain the ordering of [msg, code symbols] in the coded msg.
+  function encodeRSBlock(msg, n, k) {
+    var remainder = UTIL.polynomialDiv(msg, genPoly, n, k);
```

Comment:
That is an excellent question. I suppose we should figure that part out soon :P 
I haven't made a function to generate the polynomial yet. We could have a constant one for some n,k initially.